code/app/src/main/java/com/example/sigma_blue/activities/PhotoTakingActivity.java

```diff
@@ -0,0 +1,95 @@
+package com.example.sigma_blue.activities;
+
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.widget.ImageView;
+
+import androidx.annotation.NonNull;
+
+import com.example.sigma_blue.R;
+import com.example.sigma_blue.context.GlobalContext;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.UUID;
+
+public class PhotoTakingActivity extends BaseActivity{
+
+    static final int REQUEST_IMAGE_CAPTURE = 1;
+
+    private GlobalContext globalContext;        // Global context object
+
+    private FirebaseStorage storage = FirebaseStorage.getInstance();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState); // Activity super
+        globalContext = GlobalContext.getInstance();
+
+        // Set xml view
+        setContentView(R.layout.photo_loading_activity);
+
+        dispatchTakePictureIntent();
+
+
+
+
+    }
+
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        try {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+        } catch (ActivityNotFoundException e) {
+            // display error state to the user
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent Data) {
+
+        super.onActivityResult(requestCode, resultCode, Data);
+
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK){
+            Bundle extras = Data.getExtras();
+            Bitmap imageBitmap = (Bitmap) extras.get("data");
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
I think that a function that converts Bitmap to byte[] would be good for code quality.